subvol-mount-generator/suse-btrfs-subvol-mount-generator

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+output="${1:?}"
+
+declare -A mountpoints
+# the list of potential mount points we may need to generate a unit for
+for i in /var /usr/local /srv /root /opt /home /.snapshots; do mountpoints["$i"]=""; done
+
+generate_subvol_mounts() {
```

Comment:
There is just a single function in here and it's called immediately, so IMO this should just be top level code.